opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/malware-test.ts

```diff
@@ -0,0 +1,407 @@
+  });
+  it('should malware deleted', async () => {
+    const DELETE_QUERY = gql`
+      mutation malwareDelete($id: ID!) {
+        malwareEdit(id: $id) {
+          delete
+        }
+      }
+    `;
+    // Delete the malware
+    await queryAsAdmin({
+      query: DELETE_QUERY,
+      variables: { id: malwareInternalId },
+    });
+    // Verify is no longer found
+    const queryResult = await queryAsAdmin({ query: READ_QUERY, variables: { id: malwareStixId } });
+    expect(queryResult).not.toBeNull();
+    expect(queryResult.data?.malware).toBeNull();
+  });
+});
+
+describe('Malware createdBy relation verifications', () => {
+  let malwareWithoutCreatedById: string;
+  const malwareWithoutCreatedByIdName = 'Malware that is created without relation createdBy';
+  const malwareIdForCleanup: string[] = [];
+
+  it('should be forbidden to use an existing id (ex: User) instead of Identity id in createdBy', async () => {
+    const CREATE_QUERY = gql`
+      mutation MalwareAdd($input: MalwareAddInput!) {
+        malwareAdd(input: $input) {
+          name
+          createdBy {
+            id
+          }
+        }
+      }
+    `;
+    const MALWARE_WITH_BAD_CREATED_BY = {
+      input: {
+        name: 'Malware should not be created with a userid as CreatedBy',
+        createdBy: USER_PARTICIPATE.id,
+      },
+    };
+    const malwareCreationResponse = await queryAsAdmin({
+      query: CREATE_QUERY,
+      variables: MALWARE_WITH_BAD_CREATED_BY,
+    });
+    expect(malwareCreationResponse.errors?.length, 'Error is expected when createdBy is not Identity.').toBe(1);
+    expect(malwareCreationResponse.data?.malwareAdd, 'No data is expected when createdBy is not Identity.').toBeNull();
+  });
+
+  it('should be allowed to create a Malware with unknown createdBy', async () => {
+    const CREATE_QUERY = gql`
+      mutation MalwareAdd($input: MalwareAddInput!) {
+        malwareAdd(input: $input) {
+          name
+          createdBy {
+            id
+          }
+        }
+      }
+    `;
+    const MALWARE_WITH_NOT_EXISTING_CREATED_BY = {
+      input: {
+        name: 'Malware should be created without relation createdBy',
+        createdBy: 'this-is-an-invalid-id',
+      },
+    };
+    const malwareCreationResponse = await queryAsAdminWithSuccess({
+      query: CREATE_QUERY,
+      variables: MALWARE_WITH_NOT_EXISTING_CREATED_BY,
+    });
+    expect(malwareCreationResponse.data?.malwareAdd.createdBy).toBeNull();
+    malwareIdForCleanup.push(malwareCreationResponse.data?.malwareAdd.id);
+  });
+
+  it('should be allowed to create a Malware with legit Identity in createdBy', async () => {
+    const CREATE_QUERY = gql`
+      mutation MalwareAdd($input: MalwareAddInput!) {
+        malwareAdd(input: $input) {
+          name
+          createdBy {
+            id
+          }
+        }
+      }
+    `;
+    const MALWARE_WITH_BAD_CREATED_BY = {
```

Comment:
small typo when copy/pasting from previous test, we can name it MALWARE_WITH_VALID_CREATED_BY